AK/Statistics.h

```diff
@@ -29,6 +29,28 @@ requires(IsArithmetic<T>) class Statistics {
     T const sum() const { return m_sum; }
     float average() const { return (float)sum() / size(); }
 
+    T const min()
+    {
+        T min_v = m_values[0];
+        for (T number : values()) {
```

Comment:
I don't love that this is storing the values, especially considering that all of these values can be calculated without storing the inputs...

Reply:
You can't compute standard deviation without having the mean first, can you? You'd need to store the data for that. One could argue how useful standard deviation is, but that's another story

Reply:
You can, you just two accumulators for sum(x^2) and sum(x), and a counter for how many x's there were:
`sqrt(sum(x^2)/N - (sum(x)/N)^2)`

Reply:
If we set the min/max values after each `.add()` as [@mustafaquraish suggested](https://github.com/SerenityOS/serenity/pull/9750#issuecomment-911968642), what should be the default value for them?

Reply:
They can just be `Optional<double>`'s, no default value needed.

Reply:
Online algorithm for stddev is a nice feature, I believe if something can be calculated on `.add()` we should do it. I think, I can implement the [Welford's online algorithm](https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance) for variance, if that sounds okay. Since we are talking about multiple little tweaks and improvements, I can rename the PR or open another one. Let me know which sounds the best.

Reply:
If we're gonna do that, let's merge this and do that in another PR.
The only thing for _this_ PR would be the "let's split this up into multiple lines" bit.

Reply:
Sure! I will work on the "multiple lines" thing and after that, the other improvements.
I might be asking too many questions, sorry. I want to implement tests for `AK::Statistics`, should they be included in the "little tweaks" PR or should I send another PR after that one?

Reply:
The tests can be put in the "little tweaks" PR :)